docs/scripting.md

```diff
@@ -260,6 +260,26 @@ module.exports = (robot) ->
     res.send res.random leaveReplies
 ```
 
+## Custom Listeners
+
+While the above helpers cover most of the functionality the average user needs (hear, respond, enter, leave, topic), sometimes you would like to have very specialized matching logic for listeners. If so, you can use `listen` to specify a custom match function instead of a regular expression.
+
+The match function must return a truthy value if the listener callback should be executed. The truthy return value of the match function is then passed to the callback as response.match.
```

Comment:
Hmm, okay. The example I had in mind (factoids) felt too complicated for `scripting.md`, so I put it in `patterns.md`. Can you think of a simpler example that would fit in `scripting.md`? Alternatively, I can just move the example over, but that would leave us example-less in `patterns.md`.